### DISCUSSION TITLE
How to extract Question and Question/CheckBox values list  ,  from a pdf form along side its input fields(widget objects)

### DISCUSSION THREAD

**Discussion by 808Code** [pymupdf/PyMuPDF#3299] [Looking for help] [answered]
For example in the image : 
![UnNumbered Field Image](https://github.com/pymupdf/PyMuPDF/assets/75775205/385e1a25-578d-4a17-a1d3-f0ed624d71ab)

Is it possible to extract  : 
`
Question : Number of applicants ,
Associated Input Field : number_of_applicants_widget_object 
`
or
 
`
Question : Purpose of Loan?,
Checkboxes :  [ Construction : construction_widget_object , Refinance: refinance_widget_object , Other : other_widget_object }]
`

**Comment by JorjMcKie** [comment, marked as answer]
You can always extract page text as I am sure you know.
You could write code that searches for the question text, takes the returned rectangle and widens it to the full page width.
Then look at all field rectangles intersecting this widened rectangle and identify their field values.

**Reply by 808Code** [reply]
Ooh , That can work for a lot of cases. Thank you.
